### ISSUE TITLE
getch & get_wch won't read specific keys correctly 

### ISSUE THREAD

**Issue by GiorgosXou** [wmcbrine/PDCurses#126]
So, i am running the same examples as seen below, both on windows and linux but when i'm *(on windows and i'm)* typing one of those spesific 4 symbols `'"\|`, `get_wch` or `getch` return gibberish\nonsense *(I built the source-code from master using `make -f Makefile WIDE=Y UTF8=Y DLL=Y`)*:

  ## Example with `getch()`
```c++
#include <curses.h>

int main( void)
{
    int ch =-1;
    initscr();
    noecho ();
    keypad(stdscr, TRUE);
    
    while(ch != 'Q')
    {
        ch = getch();
        addch(ch);
    }
    endwin( );
    return(0);
}
```
i type:
```terminal
This is a test of quotation-mark symbols "' back-slash \ , straight-line |
```
it prints:
```terminal
This is a test of quotation-mark symbols 潬潃back-slash ^@ , straight-line ^@
```
  ## Example with `get_wch()`
```c++
#define PDC_WIDE TRUE
#include <curses.h>
#include <locale.h>

int main( void)
{
    setlocale(LC_ALL, "");
    wint_t  * ch = new wint_t(0);
    initscr();
    noecho ();
    keypad(stdscr, TRUE);
    
    while(*ch != 'Q')
    {
        get_wch(ch);
        const cchar_t wch = A_NORMAL|*ch; // For PDCurses
        //const cchar_t wch = {A_NORMAL,{*ch}}; // For NCurses
        add_wch(&wch);
        //printf(" %d |",*ch);
    }
    endwin( );
    return(0);
}
```
i type *(copy paste)*:
```terminal
This is a test of quotation-mark symbols "' back-slash \ , straight-line | and other random ⌥❖ ✖ ៛ ₪ ₯ ₠ ₧
```
it prints:
```terminal
This is a test of quotation-mark symbols 潬潃back-slash ^@ , straight-line ^@ and other random ⌥❖ ✖ ៛ ₪ ₯ ₠ ₧  
```

***if i remember well from old tests, this issue was not pressent with the latest realease but only appears with the up-to-date\\"master" source code, (not 100% sure though...)***. Also this issue is not present within NCurses, everything works as i would expect it to work

**Comment by GiorgosXou** [comment]
![pdcursesbug3 gif](https://user-images.githubusercontent.com/11987271/142511727-20236e7f-abd7-47a3-97ca-0a1e5c0db849.gif)

**Comment by GiorgosXou** [comment]
![pdcursesbug3 gif](https://user-images.githubusercontent.com/11987271/142511859-565cef8d-7eb5-49fc-a27d-dedb8f75ad7b.gif)

**Comment by wmcbrine** [comment]
Weird. I can confirm, with MinGW. Then I tried it with Watcom -- no problem. Then I tried it with MSVC -- pipe and backslash were fine, but the quote characters were broken. I'll keep on it...

BTW, you should start a separate issue for the transparency thing.

**Comment by GiorgosXou** [comment]
So does this mean that it might be a potential `MinGW`' & `MSVC`s bug?

**Comment by wmcbrine** [comment]
Fixed by 6c31f226922bd8653e7ed40cd4b11cee9845ea97 -- may need to explore more to understand the issue.
